src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,28 @@
+package seedu.addressbook.data.person;
+
+public abstract class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+    
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+
+    public boolean isPrivate() {
+        return isPrivate;
+    }
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
This works but note the case where the 2 contacts could be different classes but have the same value.